### PR TITLE
Add benchmarking

### PR DESCRIPTION
**Pull request by smcguire-cmu**
Update template with copier to add benchmarking, and add crossmatch benchmark

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -10,3 +10,7 @@ Welcome to lsdb's documentation!
 
    Home page <self>
    API Reference <autoapi/index>
+<<<<<<< before updating
+=======
```
Comment (delucchi-cmu) [inline docs/index.rst:14]: zoinks

File: .gitignore [hunk 2]
```diff
@@ -138,3 +138,9 @@ dask-worker-space/
 
 # tmp directory
 tmp/
+
```
Comment (delucchi-cmu) [inline .gitignore:141]: Also ignore `_results` and `_html` (though that should get fixed in the template eventually - https://github.com/lincc-frameworks/python-project-template/issues/269)

File: docs/index.rst [hunk 3]
```diff
@@ -10,3 +10,7 @@ Welcome to lsdb's documentation!
 
    Home page <self>
    API Reference <autoapi/index>
+<<<<<<< before updating
```
Comment (camposandro) [inline docs/index.rst:13]: This looks like an unresolved conflict

**Review by camposandro — CHANGES_REQUESTED** [review]

File: docs/index.rst (hunk 1 above)
Comment (smcguire-cmu) [reply docs/index.rst:14]: whoops yeah I just noticed that

File: .gitignore (hunk 2 above)
Comment (smcguire-cmu) [reply .gitignore:141]: got it

**Review by delucchi-cmu — APPROVED** [review]

**Review by camposandro — APPROVED** [review]

**Pull request merged**